File: 04_BehaviouralDesignPatterns/03_StrategyPattern/cpp/badCode.cpp
#include <iostream>
#include <string>

class RideMatchingService {
public:
    void matchRider(const std::string& riderLocation, const std::string& matchingType) {
        if (matchingType == "nearest") {
            // Logic to find the nearest driver
            std::cout << "Matching rider at " << riderLocation << " with nearest driver." << std::endl;
        }
        else if (matchingType == "fastest") {
            // Logic to find the fastest driver
            std::cout << "Matching rider at " << riderLocation << " with fastest driver." << std::endl;
        }
        else if (matchingType == "cheapest") {
            // Logic to find the cheapest driver
            std::cout << "Matching rider at " << riderLocation << " with cheapest driver." << std::endl;
        }
        else {
            std::cout << "Invalid matching type." << std::endl;
        }
    }
};

int main() {
    RideMatchingService rideMatchingService;
    rideMatchingService.matchRider("Downtown", "nearest");
    rideMatchingService.matchRider("Uptown", "fastest");
    rideMatchingService.matchRider("Suburbs", "cheapest");
    rideMatchingService.matchRider("City Center", "luxury"); // Invalid type
    
    return 0;
}

File: 04_BehaviouralDesignPatterns/03_StrategyPattern/badCode.java
class RideMatchingServic {
    public void matchRider(String riderLocation, String matchingType) {
        if (matchingType.equals("nearest")) {
            // Logic to find the nearest driver
            System.out.println("Matching rider at " + riderLocation + " with nearest driver.");
        } else if (matchingType.equals("fastest")) {
            // Logic to find the fastest driver
            System.out.println("Matching rider at " + riderLocation + " with fastest driver.");
        } else if (matchingType.equals("cheapest")) {
            // Logic to find the cheapest driver
            System.out.println("Matching rider at " + riderLocation + " with cheapest driver.");
        } else {
            System.out.println("Invalid matching type.");
        }
    }
}

public class badCode {
    public static void main(String[] args) {
        RideMatchingServic rideMatchingService = new RideMatchingServic();
        rideMatchingService.matchRider("Downtown", "nearest");
        rideMatchingService.matchRider("Uptown", "fastest");
        rideMatchingService.matchRider("Suburbs", "cheapest");
        rideMatchingService.matchRider("City Center", "luxury"); // Invalid type
    }
}